tests/end_to_end/test_helper.hpp

```diff
@@ -32,6 +32,21 @@ class _test_helper_libbpf
     _single_instance_hook* cgroup_inet4_connect_hook;
 };
 
+class _test_handle_helper
```

Comment:
This has a few issues (implicit copy constructor / assignment operator etc). Have you considered using [std::unique with a deleter](https://en.cppreference.com/w/cpp/memory/unique_ptr)?

Reply:
Fixed the copy constructor and assignment operator.

Reply:
Just set them = delete. That disables them

Reply:
Just set them = delete. That disables them

Reply:
Fixed.